addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/change/MetadataMergePomChangeListener.java

```diff
@@ -0,0 +1,175 @@
+
+    @Inject
+    @WeftManaged
+    @ExecutorConfig( daemon = true, priority = 7, named = "indy-events" )
+    private Executor executor;
+
+    public void metaClear( @Observes final FileEvent event )
+    {
+
+        final String path = event.getTransfer().getPath();
+
+        if ( event instanceof FileAccessEvent )
+        {
+            return;
+        }
+
+        if ( !path.endsWith( ".pom" ) )
+        {
+            return;
+        }
+
+        final StoreKey key = getKey( event );
+        final String versionPath = normalize( parentPath( path ) );
+        final String clearPath = normalize( normalize( parentPath( versionPath ) ), MavenMetadataMerger.METADATA_NAME );
+        try
+        {
+            if ( hosted == key.getType() )
+            {
+                HostedRepository hosted = dataManager.query().getHostedRepository( key.getName() );
+                try
+                {
+                    if ( doClear( hosted, clearPath ) )
+                    {
+                        versionMetadataCache.remove( hosted.getKey() );
+                    }
+                }
+                catch ( final IOException e )
+                {
+                    logger.error(
+                            String.format( "Failed to delete: %s from hosted: %s when pom version changed. Error: %s",
+                                           path, hosted, e.getMessage() ), e );
+                }
+
+                final Set<Group> groups = dataManager.query().getGroupsContaining( key );
+                if ( groups != null )
+                {
+                    for ( final Group group : groups )
+                    {
+                        try
+                        {
+                            if ( doClear( group, clearPath ) )
+                            {
+                                versionMetadataCache.remove( group.getKey() );
+                            }
+                        }
+                        catch ( final IOException e )
+                        {
+                            logger.error( String.format(
+                                    "Failed to delete: %s from its group: %s when pom version changed. Error: %s", path,
+                                    group, e.getMessage() ), e );
+                        }
+                    }
+                }
+            }
+        }
+        catch ( final IndyDataException e )
+        {
+            logger.warn( "Failed to regenerate maven-metadata.xml for artifacts after deployment to: {}"
+                                 + "\nCannot retrieve associated groups: {}", e, key, e.getMessage() );
+        }
+    }
+
+    private boolean doClear( final ArtifactStore group, final String path )
+            throws IOException
+    {
+        boolean isCleared = false;
+        logger.trace( "Updating merged metadata file: {} in group: {}", path, group.getKey() );
+
+        final Transfer[] toDelete = { fileManager.getStorageReference( group, path ),
```

Comment:
Will TransferDecorator not take care of deleting the derivative files? I thought we had a decorator that handled this.

Reply:
seem for group meta: .info will not be deleted for any trigger, other files will be deleted indeed by the decorators, simplify this in following commit.